manifests/plugin/openscap.pp

```diff
@@ -0,0 +1,63 @@
+# = Foreman Proxy Openscap plugin
+#
+# This class installs openscap plugin
+#
+# === Parameters:
+#
+# $enabled::                    enables/disables the plugin
+#                               type:boolean
+# $configure_openscap_repo::    Enable custom yum repo with packages needed for foreman_openscap,
+#                               type:boolean
+class foreman_proxy::plugin::openscap (
+  $enabled                 = $::foreman_proxy::plugin::openscap::params::enabled,
+  $configure_openscap_repo = $::foreman_proxy::plugin::openscap::params::configure_openscap_repo,
+) inherits foreman_proxy::plugin::openscap::params {
+  validate_bool($enabled)
+  validate_bool($configure_openscap_repo)
+
+  case $::osfamily {
+    'RedHat': {
+
+      if $configure_openscap_repo {
+        case $::operatingsystem {
+          'fedora': {
+            $repo = 'fedora'
+          }
+          default: {
+            $repo = 'epel'
+          }
+        }
+
+        if $::osreleasemajor == undef {
+          if $::operatingsystemmajrelease == undef {
+            $versions_array = split($::operatingsystemrelease, '\.') # facter 1.6
+            $major = $versions_array[0]
+          } else {
+            $major = $::operatingsystemmajrelease # facter 2.2
+          }
+        } else {
+          $major = $::osreleasemajor
+        }
+
+        yumrepo { 'isimluk-openscap':
```

Comment:
Packages should be submitted to our repos via https://github.com/theforeman/foreman-packaging/tree/rpm/develop